src/syngen/ml/vae/models/dataset.py

```diff
@@ -417,13 +420,12 @@ def _avro_data_pipeline(self, df, schema):
         self.float_columns = self.float_columns - self.categ_columns - self.binary_columns
         self.str_columns = set(column for column, data_type in schema.items() if data_type == 'string')
         self.categ_columns -= self.long_text_columns
-        self.str_columns = \
-            self.str_columns - self.categ_columns - self.int_columns - self.binary_columns - self.long_text_columns
-        self.date_columns = \
-            get_date_columns(df, list(self.str_columns)) - self.categ_columns - \
-            self.binary_columns - self.long_text_columns
+        self.str_columns = self.str_columns - self.categ_columns - self.binary_columns \
+                           - self.long_text_columns - self.uuid_columns
+        self._set_date_columns(df)
```

Comment:
What's the purpose of the second function execution here? The first one was on line 402 in the new code.

Reply:
There are 2 different pipelines - _general_data_pipeline, _avro_data_pipeline